content/en/apps/guides/integrations/best-practices/rapidpro.md

```diff
@@ -4,106 +4,105 @@ linkTitle: "RapidPro"
+   1. Avoid the Android Channel going to sleep 
+    The following apply (based on Samsung A7 2018) with more details available on [dontkillmyapp](https://dontkillmyapp.com/samsung):
+      - Android Settings > Apps > RapidPro > Battery > Optimize battery usage > Don't Optimize.
+      - Android Settings > Device care > Battery > App power management > Apps that won’t be put to sleep. Make sure that the app is in this list. Turn off "Put unused apps to sleep."
+      - Android Settings > System > Developer Options. Background Limit is set to Standard Limit. Make sure the RapidPro app has the toggle enabled in "Background Check" and ALL of my apps have "App Standby state: ACTIVE" in Standby apps.
+      - Keep the phone plugged into a power source.
+
+   2. Automatically Wake up the Phone 
+      - The production RapidPro system uses Google Cloud Messaging which wakes up the phone whenever a message is sent. Install an APK that makes the channel to relay messages as soon as RapidPro emits.
+   3. Alerts for when things go wrong 
+      - Send monitoring emails and alert various parties when the Android Channel goes to sleep or becomes unavailable. This can be done from the channel's management page under `Alert Email`
+
+- The Android channels can however, be used with a [bulk sender](https://help.nyaruka.com/en/articles/5153032-using-a-bulk-sender) to get past the 330 outgoing messages per hour.
+- We recommend that you use shortcodes for SMS messaging, which can either be shared or dedicated. Dedicated are preferred because recipients do not have to include the keyword with each response submitted. Make sure to make arrangements for the shortcode in advance to avoid lags due to the duration it might take to fully set up one.
+- Regarding SMS costs from experience, zero-rated messages are recommended so that respondents do not incur charges.
+
+### Flow design
+Tips and best practices are listed below
+ - Length of the survey.
+   - If possible, keep the questions to between 5 and 10. Long surveys may either lead to low completion rates or rushed responses that affects the data quality.
+- Types of questions
+   - Close ended questions are better and easier to respond and handle in RapidPro since respondents only have to send a number or letter such as 1 for “Yes”, 2 for “No”. Avoid sensitive questions since privacy cannot be guaranteed where shared phones are common.
+- Intro and outro messages
+   - These serve the purpose of giving the survey details such as background of the survey, number of questions, data protection, whether there shall be follow up, SMS billing, thank you notes etc. Include these if possible.
+- Consent and opt out
+   - Include questions that give the respondent an opportunity to opt in or out of the survey. If they opt out, do not send a follow up text.
+- SMS length
+   - Keep it short, to the size of one SMS (160 characters). Longer messages may not display well on the recipients’ devices.
```

Comment:
worth noting message length even for each translation.

how do we handle long names if the name is part of the message? this can go over character limit and i think i've seen us truncate?